examples/java/build.gradle

```diff
@@ -64,15 +64,18 @@ dependencies {
     // but needed for compile only under Java21, specifically, required for extending from AbstractKafkaAvroDeserializer
     compileOnly library.java.kafka
   }
+  if (JavaVersion.current().compareTo(JavaVersion.VERSION_11) >= 0) {
```

Comment:
This if clause does not change anything as JavaVersion.VERSION_11 is always >=1 for build system for now. Java8 tests are run on forkOptions (set test to use Java8 while Gradle still run on Java11)

Reply:
This could help for Beam CI test, but not affect published artifacts (they still get the dependency regardless of Java version), and Beam jars still does not compile as long as bytecode version is set to 8 (see the failed tests below)